module/zfs/zpl_file.c

```diff
@@ -395,9 +397,9 @@ zpl_iter_write(struct kiocb *kiocb, struct iov_iter *from)
 	 */
 	ret = generic_write_checks(kiocb, from);
 	count = ret;
```

Comment:
You could move the `count = ret;` assignment after the ret check now, but it really doesn't matter.